### PR TITLE
Add k3s deployment

### PR DESCRIPTION
**Pull request by joshuai96**
Add an example k3s deployment for a single host (bare metal or vm).

### CODE REVIEW

**Comment by joshuai96** [comment]
@fkr as requested. I cleaned up the documentation a bit and added an overview diagram.

**Comment by joshuai96** [comment]
We talked about configuring the domains needed by the deployments, this requires some more tooling for templating, like `helm`.

For now this is not included, but I will create an issue for this.

**Comment by joshuai96** [comment]
Blocks #8, as main didn't include the k3s deployment.

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Dex needs a GitHub Applications Client Secret in `dex/dex-secrets.env`, please refer to `dex/dex-secrets-example.env`.
+
+Set the `issuer` and `redirectURI` at `dex/config.yaml` to your domain. Keep in mind, that dex needs it's own domain or subdomain.
+
+Other Dex related configuration is located in `dex/dex.env`, `web/web-secrets.env` and `web/web.env` to fill the configuration template `dex/config.yaml`.
+
+## Oathkeeper
+
+Set your domain in `oathkeeper/config.yaml` in the `authenticators`s `jwks_urls` and `trusted_issuers` to point towards Dex.
```
Comment (joshmue) [inline docs/configuration.md]:
```suggestion
Set your domain in `oathkeeper/config.yaml` in the `authenticators`' `jwks_urls` and `trusted_issuers` to point towards Dex.
```

?

Or more explicit:

```suggestion
Set your domain in `oathkeeper/config.yaml` at `authenticators.jwt.config.jwks_urls` and `authenticators.jwt.config.trusted_issuers` to point towards Dex.
```

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Web frontend
+
+In `web/web.env` configure the OIDC authentication callback and the API url. The API url must be pointing to the external domain, not the k8s service name.
```
Comment (joshmue) [inline docs/configuration.md]:
```suggestion
In `web/web.env` configure the OIDC authentication callback and the API url. The API URL must be pointing to the external domain, not the K8s service name.
```

File: docs/k3s.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## k3s setup
+
+Setting up k3s to run k8s workloads on the machine.
+
+### Firewall settings
+
+Allow communication to and from the cluster.
+
+Incomming:
```
Comment (joshmue) [inline docs/k3s.md]:
```suggestion
Incoming:
```

File: docs/k3s.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+```bash
+curl -sfL https://get.k3s.io | sh -
+```
+
+## Deployment
+
+Deploy services to the cluster, to get the status page running.
+
+If you don't have access to a domain name, consider using [nip.io](https://nip.io/) to use as (sub)domains for Dex and Oathkeepr (API server).
```
Comment (joshmue) [inline docs/k3s.md]:
```suggestion
If you don't have access to a domain name, consider using [nip.io](https://nip.io/) to use as (sub)domains for Dex and Oathkeeper (API server).
```

**Review by joshmue — COMMENTED** [review]
Just some nitpicking.

**Review by joshmue — APPROVED** [review]
LGTM

**Review by fkr — APPROVED** [review]
LGTM! For technical details, I refer to @joshmue

**Review by joshmue — APPROVED** [review]
Changes LGTM, too.

File: docs/configuration.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+In `web/web.env` configure the OIDC authentication callback and the API url. The API URL must be pointing to the external domain, not the K8s service name.
+
+## Ingress
+
+In `ingress.yaml` set your domains for Dex, Oathkeeper and the web frontend respectivly. Oathkeeper acts as the auth proxy for the API server. Exposing the API server directly, opens up the possibilty of unsupervised write actions.
+
+## Issuer
+
+Set the e-mail address in [`issuer.yaml`](issuer.yaml) to your desired e-mail address.
```
Comment (joshmue) [inline docs/configuration.md]:
```suggestion
Set the e-mail address in `issuer.yaml` to your desired e-mail address.
```

**Pull request merged**